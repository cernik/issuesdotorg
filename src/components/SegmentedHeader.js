// @flow
import React from 'react';
import {View, StyleSheet} from 'react-native';
import {noop} from '../utils/constants';
import {Styles} from '../utils/styles';
import type {ItemStateType} from '../types';
import SegmentedHeaderButton from './SegmentedHeaderButton';

const BUTTON_TYPE = {
  OPEN: 'open',
  CLOSED: 'closed',
  ALL: 'all',
};

type SegmentedHeaderProps = {
  value: ItemStateType,
  onChange: (value: ItemStateType) => void,
};

const SegmentedHeader = ({
  value = 'closed',
  onChange = noop,
}: SegmentedHeaderProps) => (
  <View style={Styles.row}>
    {Object.keys(BUTTON_TYPE).map((k, i, arr) => (
      <SegmentedHeaderButton
        key={k}
        style={
          !i
            ? styles.buttonStart
            : i === arr.length - 1
            ? styles.buttonEnd
            : null
        }
        selected={value === BUTTON_TYPE[k]}
        onPress={() => onChange(BUTTON_TYPE[k])}
        value={BUTTON_TYPE[k]}
      />
    ))}
  </View>
);

const styles = StyleSheet.create({
  buttonStart: {
    borderLeftWidth: 1,
    borderRightWidth: StyleSheet.hairlineWidth,
    borderTopLeftRadius: 8,
    borderBottomLeftRadius: 8,
  },
  buttonEnd: {
    borderRightWidth: 1,
    borderLeftWidth: StyleSheet.hairlineWidth,
    borderTopRightRadius: 8,
    borderBottomRightRadius: 8,
  },
});

export default SegmentedHeader;
