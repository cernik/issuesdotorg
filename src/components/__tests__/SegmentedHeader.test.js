// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import SegmentedHeader from '../SegmentedHeader';

describe('snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<SegmentedHeader value={'closed'} onChange={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
