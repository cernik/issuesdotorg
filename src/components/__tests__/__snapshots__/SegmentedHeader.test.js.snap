// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderTopWidth": 1,
          "minWidth": 90,
          "padding": 4,
          "paddingHorizontal": 16,
        },
        Object {
          "backgroundColor": "white",
        },
        Object {
          "borderBottomLeftRadius": 8,
          "borderLeftWidth": 1,
          "borderRightWidth": 0.5,
          "borderTopLeftRadius": 8,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 17,
              "fontWeight": "500",
            },
            Object {},
          ]
        }
      >
        open
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderTopWidth": 1,
          "minWidth": 90,
          "padding": 4,
          "paddingHorizontal": 16,
        },
        Object {
          "backgroundColor": "gainsboro",
        },
        null,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 17,
              "fontWeight": "500",
            },
            Object {
              "fontWeight": "bold",
            },
          ]
        }
      >
        closed
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderTopWidth": 1,
          "minWidth": 90,
          "padding": 4,
          "paddingHorizontal": 16,
        },
        Object {
          "backgroundColor": "white",
        },
        Object {
          "borderBottomRightRadius": 8,
          "borderLeftWidth": 0.5,
          "borderRightWidth": 1,
          "borderTopRightRadius": 8,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 17,
              "fontWeight": "500",
            },
            Object {},
          ]
        }
      >
        all
      </Text>
    </View>
  </View>
</View>
`;
